pgx/src/lib.rs

```diff
@@ -319,4 +352,12 @@ macro_rules! pg_sql_graph_magic {
 #[allow(unused)]
 pub fn initialize() {
     pg_sys::panic::register_pg_guard_panic_hook();
+    match Lazy::force(&UTF8DATABASE) {
+        Utf8Compat::Maybe => notice!("database encoding is not validated as UTF-8"),
+        Utf8Compat::Ascii => {
+            warning!("database encoding is not UTF-8, extension will accept ASCII only")
+        }
```

Comment:
These various encodings include a few which may share a few codepoint encodings with UTF-8, but the decision to only trust ASCII data from them is because that check is simpler ("high bit set? yes? okay!") and the shared codepoints are more incidental than by design.